### DISCUSSION TITLE
combined image sampler support

### DISCUSSION THREAD

**Discussion by castano** [shader-slang/slang#5803] [General]
When targeting spirv and binding the texture and the sampler to the same index produces the following warning:

```
warning 39001: explicit binding for parameter 'input_sampler' overlaps with parameter 'input_texture'
```

Is there any way to support combined image samplers as in DXC? Are there any plans for vk::combinedImageSampler support?

https://github.com/microsoft/DirectXShaderCompiler/wiki/Vulkan-combined-image-sampler-type

**Comment by wrapperup** [comment]
The `Sampler*` types like `Sampler2D` are the combined image + sampler type, and it has the same variants as `Texture*` does. With that you only need one binding, eg.

```hlsl
[[vk::binding(0, 0)]] Sampler2D input;
```

**Reply by castano** [reply]
I see. I'm trying to compile an existing HLSL code base using Slang and migrate it incrementally. Is it possible to retrieve the Texture2D and SamplerState objects associated to the Sampler2D object so that I can keep using the existing code?

**Reply by wrapperup** [reply]
Yeah you can, though it is fairly hidden from the docs and only used internally.
```
Texture2D texure = input.__getTexture();
SamplerState sampler = input.__getSampler();
```

**Reply by jkwak-work** [reply]
I think the answer is good.
Please let us know if it didn't work.
